tests/genomics/antismash/test_antismash_downloader.py

```diff
@@ -43,3 +45,27 @@ def test_error_nonempty_path(self, tmp_path):
             download_and_extract_antismash_data(self.antismash_id, tmp_path,
                                                 tmp_path / "extracted")
         assert "Nonempty directory" in e.value.args[0]
+
+    def test_nonexisting_id(self, tmp_path):
+        nonexisting_antismash_id = 'nonexisting_id'
+        download_root = tmp_path / "download"
+        download_root.mkdir()
+        extract_root = tmp_path / "extracted"
+        extract_root.mkdir()
+        with pytest.raises(urllib.error.HTTPError):
+            download_and_extract_antismash_data(nonexisting_antismash_id, download_root,
+                                                extract_root)
+        extracted_folder = extract_root / "antismash" / nonexisting_antismash_id
+        assert not extracted_folder.exists()
```

Comment:
```suggestion
```

This test could be removed. Anstismash return 404 anyway not matter the ID is non-existent or it's broken.